### PR TITLE
refactor(game): migrate primary metadata to blade

### PR DESCRIPTION
**Pull request by wescopeland**
This PR is part of a series of PRs to break hub pages away from game pages.

In this PR, the primary meta components of game pages have been refactored to Blade to allow for easier reuse without the need for lots of code duplication.

The intention of this PR is to introduce no functional changes whatsoever. It is just a migration of the existing primary meta components to Blade.

This PR will very likely conflict with https://github.com/RetroAchievements/RAWeb/pull/1500. If this PR is merged first, we can work together to resolve the conflicts in @EmoonX's. If the other PR is merged first, we can resolve the conflicts here.

Screenshot 2023-05-07 at 2 05 11 PM

### CODE REVIEW

File: resources/views/platform/components/game/primary-meta.blade.php [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+    'imageIcon' => '#',
+])
+
+<div class="flex flex-col sm:flex-row sm:w-full gap-x-4 gap-y-2 items-center mb-4">
+    <img 
+        class="aspect-1 object-cover rounded-sm w-[96px] h-[96px]" 
+        src="{{ $imageIcon }}"
+        width="96" 
+        height="96" 
+        alt="Game or hub icon"
```
[the remark this answers is not in the document]
Comment (wescopeland) [reply resources/views/platform/components/game/primary-meta.blade.php]: Good idea. To keep the component as dumb as possible, what I've done is added an additional prop called `metaKind` which can be driven by `$isFullyFeaturedGame` (or anything else we want).

File: public/gameInfo.php [hunk 2]
```diff
@@ -685,32 +685,51 @@ function ResetProgress() {
...
+                        <?= RenderMetadataTableRow("Developer", $developer, $gameHubs, ["Hacker"]); ?>
+                        <?= RenderMetadataTableRow("Publisher", $publisher, $gameHubs, ["Hacks"]); ?>
+                        <?= RenderMetadataTableRow("Genre", $genre, $gameHubs, ["Subgenre"]); ?>
+                    @else
+                        <?= RenderMetadataTableRow("Developer", $developer); ?>
+                        <?= RenderMetadataTableRow("Publisher", $publisher); ?>
+                        <?= RenderMetadataTableRow("Genre", $genre); ?>
+                    @endif
+
+                    <?= RenderMetadataTableRow("Released", $released); ?>
```
[the remark this answers is not in the document]
Comment (wescopeland) [reply public/gameInfo.php]: Probably not without it being a mess. However this seems like a good opportunity to refactor `RenderMetadataTableRow()`, so I've done just that in https://github.com/RetroAchievements/RAWeb/pull/1566/commits/6d4b004652cffe2b1a4433d73e1ad6702777f502. This is now a Blade component (though still rendered in the slot of `<x-game.primary-meta />`). 

I am indifferent on whether or not we should get rid of the slot and have the rows rendered directly in the primary meta component. On one hand, not doing that keeps the component fairly dumb. On the other, having a moderately-sized slot block might be kind of ugly. I will defer to your judgment.

File: resources/views/platform/components/game/heading.blade.php [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+    'iconUrl' => asset("assets/images/system/unknown.png")
+])
+
+<?php
+// TODO: Migrate renderGameTitle to a Blade component.
+$renderedTitle = renderGameTitle(e($gameTitle));
+?>
+
+<h1 class="text-h3">
+    <span class="block mb-1">{{ $renderedTitle }}</span>
```
Comment (Jamiras) [inline resources/views/platform/components/game/heading.blade.php:13]: This doesn't work for decorated titles.

image

image

Comment (wescopeland) [reply resources/views/platform/components/game/heading.blade.php]: Resolved by https://github.com/RetroAchievements/RAWeb/pull/1566/commits/f35e86ed06be77c29c15493479320ba9fc098833.

**Review by Jamiras — APPROVED** [review]

**Pull request merged**